src/NHibernate/Proxy/ProxyBuilderHelper.cs

```diff
@@ -101,11 +102,19 @@ internal static IEnumerable<MethodInfo> GetProxiableMethods(System.Type type)
 
 		internal static IEnumerable<MethodInfo> GetProxiableMethods(System.Type type, IEnumerable<System.Type> interfaces)
 		{
-			var proxiableMethods =
-				GetProxiableMethods(type)
+			if (type.IsInterface || type == typeof(object) || type.GetInterfaces().Length == 0)
```

Comment:
Are you sure it's valid shortcut?  I think `type.IsInterface`  part should be removed. In theory interface can contain multiple interfaces with the same members:
```C#
interface IReadOnlyId
{
	int Id { get; }
}

interface IId
{
	int Id { get; set; }
}

//IFoo contains both IReadOnlyId.Id and IId.Id getters. It's ambiguous but valid code
interface IFoo : IReadOnlyId, IId
{
}

public class Foo : IFoo
{
	public virtual int Id { get; set; }
}
```
So for `IFoo` proxy (not sure about `Foo` proxy) duplicated methods are still generated.
Also I wonder when  interface `IFoo` is used as proxy will both `IId.Id` and `IReadOnlyId.Id` access for entity  work without proxy initialization.

Reply:
The type there is "Foo" (a class), so this check does not apply.

Reply:
If you use "IFoo" as a base for proxy. It would generate duplicate methods because all interface method are generated explicitly unless they are implemented implicitly in the base class.

In short - add tests to see the behaviour.

Reply:
>The type there is "Foo

My concern is mainly for interface proxy - `IFoo`. I thought it was clear .

> unless they are implemented implicitly in the base class.

So it seems for interface proxy we should generate all members implicitly. That will solve possible duplicates and unnecessary proxy initialization.

Reply:
Added test case in #2066. Current behavior leads to unnecessary proxy initialization. But as it's different interface members it's not really duplicate methods. So in the end seems unrelated to this PR.